game.py

```diff
@@ -1,232 +1,265 @@
-            self.sqr_grid[prev_x][prev_y].draw_colored_rect(BLACK)
-            grid_x = cords[0]
-            grid_y = cords[1]
-            self.sqr_grid[grid_x][grid_y].draw_colored_rect(color)
-            prev_x = grid_x
-            prev_y = grid_y
-
-    def draw_grid(self, new):
-        if new:
-            row = 0
-            for x in range(OFFSET, WINDOW_WIDTH, BLOCKSIZE):
-                col = 0
-                for y in range(OFFSET, WINDOW_HEIGHT, BLOCKSIZE):
-                    rect = ColoredRect(WHITE, x, y)
-                    self.sqr_grid[row][col] = rect.draw_colored_rect(WHITE, 1, False)
-                    col += 1
-                row += 1
-        else:
-            for row in self.sqr_grid:
-                for rect in row:
-                    rect.draw_colored_rect(WHITE, 1, False)
-
-    def draw_future_grid(self, colors):
-        for i, color in enumerate(colors):
-            self.next_sqrs = ColoredRect(
-                color,
-                BLOCKSIZE,
-                ((i + 5) * BLOCKSIZE) +
-                (i * 25)
-            ).draw_colored_rect(color)
-
-    def draw_predicted(self, next_colors):
-        placed = 0
-        while True:
-            if placed == 3:
-                break
-            x = random.randint(150, 600)
-            y = random.randint(150, 600)
-            x, y = normalize_cords(x, y)
-            x_grid = int((x / 50) - 3)
-            y_grid = int((y / 50) - 3)
-            if self.space[x_grid][y_grid] != 0:
-                continue
-            color = next_colors.pop()
-            self.sqr_grid[x_grid][y_grid] = ColoredRect(
-                color,
-                x,
-                y
-            ).draw_colored_rect(color)
-            self.space[x_grid][y_grid] = 1
-            placed += 1
-
-
-if __name__ == '__main__':
-    AlignIt()
+from __future__ import annotations
+
+import random
+import sys
+from time import sleep
+
+import pygame
+
+from astar import astar
+from coloredRect import ColoredRect
+
+
+WINDOW_HEIGHT = 600
+WINDOW_WIDTH = 600
+OFFSET = 150
+BLOCKSIZE = 50
+BLACK = (0, 0, 0)
+GREY = (160, 160, 160)
+WHITE = (200, 200, 200)
+GREEN = '#00A14B'
+RED = '#ED1C24'
+BLUE = '#21409A'
+YELLOW = '#FFDE17'
+PURPLE = '#7F3F98'
+CYAN = '#009599'
+PINK = (255, 153, 255)
+BROWN = (255, 128, 0)
+ORANGE = '#F26522'
+colors = [
+    GREEN, RED, BLUE, YELLOW, PURPLE, CYAN, ORANGE, BROWN,
+]
+
+
+def rand_color():
+    return random.choice(colors)
+
+
+def normalize_cords(x, y):
+    x = x - (x % 50)
+    y = y - (y % 50)
+    return (x, y)
+
+
+class AlignIt:
+    dim = 9
+
+    def __init__(self):
+        self.sqr_grid = [[0 for _ in range(self.dim)] for _ in range(self.dim)]
+        self.space = [[0 for _ in range(self.dim)] for _ in range(self.dim)]
+        self.next_sqrs = []
+        self.score = 0
+        self.selected_square = None
+        self.grow = True
+        self.move_made = True
+        self.same_color_counter = 0
+
+        self.main()
```

Comment:
Since `self.main()` is the last statement in the method, there's no harm in doing that.
I would have agreed with the bot if it just suggested moving the call to `self.main()` to outside the `__init__` method - it does look wrong there indeed. But it doesn't have to do with the object not being fully initialized.